src/braket/circuits/circuit.py

```diff
@@ -1116,14 +1129,20 @@ def to_ir(
         if ir_type == IRType.JAQCD:
             return self._to_jaqcd()
         elif ir_type == IRType.OPENQASM:
-            if serialization_properties and not isinstance(
-                serialization_properties, OpenQASMSerializationProperties
-            ):
-                raise ValueError(
-                    "serialization_properties must be of type OpenQASMSerializationProperties "
-                    "for IRType.OPENQASM."
+            if recompute or not self._cached_openqasm:
```

Comment:
nit: imo this would be more readable if the condition was `self._cached_openqasm and not recompute`